eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,275 @@
+        $savePropertiesScriptPath = Join-Path -Path $PSScriptRoot "Save-Package-Properties.ps1"
+        & $savePropertiesScriptPath -serviceDirectory $ServiceDirectory -outDirectory $PackageInfoDirectory
+      }
+    }
+
+    $continueValidation = $false
+    if ($packages)
+    {
+      foreach($pkgPath in $packages.Values)
+      {
+        $pkgPropPath = Join-Path -Path $PackageInfoDirectory "$PackageName.json"
+        if (-Not (Test-Path $pkgPropPath))
+        {
+            Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Package property file path $($pkgPropPath) is invalid."
+            continue
+        }
+        # Get package info from json file
+        $pkgInfo = Get-Content $pkgPropPath | ConvertFrom-Json
+        $version = [AzureEngSemanticVersion]::ParseVersionString($pkgInfo.Version)
+        if ($null -eq $version)
+        {
+            LogError "ServiceDir:$ServiceDirectory, Version info is not available for package $PackageName, because version '$(pkgInfo.Version)' is invalid. Please check if the version follows Azure SDK package versioning guidelines."
+            exit 1
+        }
+
+        Write-Host "Version: $($version)"
+        Write-Host "SDK Type: $($pkgInfo.SdkType)"
+        Write-Host "Release Status: $($pkgInfo.ReleaseStatus)"
+
+        # Ignore the validation if the package is not GA version
+        if ($version.IsPrerelease) {
+          Write-Host "ServiceDir:$ServiceDirectory, Package $($parsedPackage.PackageId) is marked with version $($parsedPackage.PackageVersion), the version is a prerelease version and the validation of spec location is ignored."
+          exit(0)
+        }
+        $continueValidation = $true
+      }
+    }
+    if($continueValidation -eq $false) {
+      Write-Host "ServiceDir:$ServiceDirectory, no package info is found for package $PackageName, the validation of spec location is ignored."
+      exit(0)
+    }
+  }
+  catch {
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to retrieve package and validate package version with exception:`n$_ "
+    exit(1)
+  }
+}
+
+try{
+  # Verify package version is not a prerelease version, only continue the validation if the package is GA version
+  Verify-PackageVersion
+
+  $ServiceDir = Join-Path $RepoRoot 'sdk' $ServiceDirectory
+  $PackageDirectory = Join-Path $ServiceDir $PackageName
+  Push-Location $PackageDirectory
+
+  # Load tsp-location.yaml if existed
+  $tspLocationYamlPath = Join-Path $PackageDirectory "tsp-location.yaml"
+  $autorestMdPath = Join-Path $PackageDirectory "src/autorest.md"
+  $swaggerReadmePath = Join-Path $PackageDirectory "swagger/README.md"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    # typespec scenario
+    $tspLocationYaml = Get-Content -Path $tspLocationYamlPath -Raw | ConvertFrom-Yaml
+    Verify-TspLocation $tspLocationYaml
+  }
+  elseif ($Language -eq "dotnet") {
```

Comment:
Better to use existence of file `autorest.md` as if-condition, not the language name for extensibility in future.

Reply:
I prefer to split the cases by languages. Go language use 'autorest.md' but we will ignore it.